From 8debe92793543401dae5d891894ccdfaa6d0bb8b Mon Sep 17 00:00:00 2001
From: Piotr Karbowski <piotr.karbowski@gmail.com>
Date: Sun, 6 Mar 2011 19:59:06 +0100
Subject: [PATCH] Mount vfat with flush option.

The "flush" mount option ensures that bytes written to files on a mounted filesystem have been flushed to the device by the time close(2) returns. 
Useful for fat32-based USB storage.

---
 src/fs.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/fs.c b/src/fs.c
index 65ea0c7..064fb60 100644
--- a/src/fs.c
+++ b/src/fs.c
@@ -21,7 +21,7 @@
 static struct FS supported_fs[] = {
     { "udf", "nosuid,nodev,user", 1, "000", ",iocharset=%s" },
     { "iso9660", "nosuid,nodev,user", 1, NULL, ",iocharset=%s" },
-    { "vfat", "nosuid,nodev,user,quiet,shortname=mixed", 1, "077", 
+    { "vfat", "flush,nosuid,nodev,user,quiet,shortname=mixed", 1, "077", 
       ",iocharset=%s",",fmask=%04o,dmask=%04o"},
     { "hfsplus", "nosuid,nodev,user", 1, NULL, 0 },
     { "hfs", "nosuid,nodev,user", 1, "077", NULL, 
-- 
1.7.4.1

